mix.exs

```diff
@@ -30,35 +30,43 @@ defmodule Bow.Mixfile do
 
   defp applications(:test) do
     [
-      :ecto, :postgrex,     # Bow.Ecto
-      :erlexec,             # Bow.Exec
-      :hackney, :sweet_xml  # Bow.Storage.S3
+      # Bow.Ecto
+      :ecto,
+      :postgrex,
+      # Bow.Exec
+      :erlexec,
+      # Bow.Storage.S3
+      :hackney,
+      :sweet_xml
     ]
   end
+
   defp applications(_), do: []
 
   defp deps do
     [
-      {:plug,     "~> 1.0"},
-      {:tesla,    "~> 0.7", github: "teamon/tesla", branch: "master"},
-
-      {:ecto,       ">= 2.0.0 and < 2.2.0", optional: true},
-      {:erlexec,    "~> 1.7.0", optional: true},
-      {:ex_aws,     "~> 2.0", optional: true},
-      {:ex_aws_s3,  "~> 2.0", optional: true},
-      {:sweet_xml,  "~> 0.6.5", optional: true},
+      {:plug, "~> 1.0"},
+      {:tesla, "~> 0.7", github: "teamon/tesla", branch: "master"},
+      {:ecto, "~> 3.2.0", optional: true},
```

Comment:
Isn't `~> 3.2.0` too restrictive for a library? Maybe just `>= 3.2.0`?

Reply:
If you are doing `>=` you should also add `< 4`. With `~>` you can do `3.2` and it should accept patch releases

Reply:
`~> 3.2` sounds responsible

Reply:
https://hexdocs.pm/elixir/Version.html

Ok. It works a bit differently than I thought. `~> 3.2` == `>= 3.2.0 and < 4.0.0` where `~> 3.2.0` == `>= 3.2.0 and < 3.3.0`

I am ok with both of them.

Reply:
I'd also prefer to choose `~> 3.2`, but the Ecto change that broke the old solution (i.e. extension of Ecto.Type behaviour definition) happened between `3.1.0` and `3.2.0`. That's why I went for `~> 3.2.0` here. No strong feelings about it though 🤔 . To be fair, it's just an `optional` dependency that's used purely for testing, so it does not affect library users.

Reply:
Changed to `~> 3.2`